tools/lint.py

```diff
@@ -60,23 +36,11 @@ def run_tool(self, name: str, *args: str) -> subprocess.CompletedProcess[bytes]:
         arg_list = list(args)
         arg_list.extend(self.folders)
 
-        run = subprocess.run(
-            arg_list, stdin=subprocess.DEVNULL, capture_output=self.is_ci, check=False
-        )
+        run_result = run_utility(name, arg_list, self.is_ci)
 
-        if run.returncode:
-            self.success = False
+        self.success = run_result["success"]
```

Comment:
This is technically a change in behaviour -- previously, we could never go from false to true, but this permits this.
When fiddling locally, I used
```python
self.success = self.success and (run_result.returncode == 0)
```
(I also took out the typed dict with the `success` property and just returned the completed process directly).

That said, looking at this function, there is the possibilty of extracting an abstract class from thsese two, so that success logic isn't duplicated...